kratos/includes/data_communicator.h

```diff
@@ -313,6 +313,10 @@ class KRATOS_API(KRATOS_CORE) DataCommunicator
     KRATOS_DEPRECATED_MESSAGE("Calling serialization-based communication for a simple type. Please implement direct communication support for this type.")
     void CheckSerializationForSimpleType(const T& rSerializedType, TypeFromBool<false>) const {}
 
+    template<bool value>
+    KRATOS_DEPRECATED_MESSAGE("Calling serialization-based communication for type bool. Communicating bools is generally problematic due to the nature of std::vector<bool>. It is recommended to cast them to ints for communication purposes.")
+    void CheckSerializationForSimpleType(const std::vector<bool>& rSerializedType, TypeFromBool<value>) const {}
```

Comment:
hm so now it uses serialization for bool?
I think we can live with this, we only need to take out the message for bool (as bools can only be exchanges with serialization) then we can keep the support for bool and no longer have the warning

Reply:
Yeah it falls back to serialization by default.

Ok, so I'll remove the warning for bool.

Reply:
yes that makes sense to me